packages/mediapipe-core/lib/src/ffi_utils.dart

```diff
@@ -6,73 +6,131 @@ import 'dart:ffi';
 import 'dart:typed_data';
 import 'package:ffi/ffi.dart';
 
-/// Converts a list of Dart strings into their C memory equivalent.
-///
-/// See also:
-///  * [prepareString]
-Pointer<Pointer<Char>> prepareListOfStrings(List<String> values) {
-  final ptrArray = calloc<Pointer<Char>>(values.length);
-  for (var i = 0; i < values.length; i++) {
-    ptrArray[i] = prepareString(values[i]);
-  }
-  return ptrArray;
+/// Offers convenience and readability extensions for detecting null pointers.
+extension NullAwarePtr on Pointer? {
+  bool get _isNull => this == null;
+
+  bool get _isNotNull => !_isNull;
+
+  /// Returns true if this is a `nullptr`.
+  bool get isNullPointer => _isNotNull && this!.address == nullptr.address;
+
+  /// Returns true if this is not a `nullptr`, but also not [null].
+  bool get isNotNullPointer => _isNotNull && this!.address != nullptr.address;
+
+  /// Returns true if this is neither [null] nor a `nullptr`.
+  bool get isNotNullAndIsNotNullPointer => _isNotNull && isNotNullPointer;
+
+  /// Returns true if this is either [null] or is a `nullptr`.
+  bool get isNullOrNullPointer => _isNull || isNullPointer;
```

Comment:
In my experience we only really use `nullptr`, we don't really have `null` for `Pointer` objects in code using `dart:ffi`. Basically `dart:ffi` always returns you a `Pointer` object, never a `Pointer?`. With what kind of code patterns do you end up with `Pointer?`?